### PR TITLE
Python: add new samples and fix streaming tool call FunctionCallContent formation

### LINKED ISSUE #6106: Python: streaming tool calls with multiple indices are broken
**Describe the bug**

When using function calling and streaming responses, the arguments from a non-zero index (aka a second or third tool call ID) are being added to the first tool call ID, showing arguments like `{}{"city": "Boston"}`. 

The issue looks to be located in the `StreamingChatMessageContent` __add__ override method, where once the FunctionCallContent is added for the first id, the enumerate loop has a break and subsequent tool call IDs don't have a chance to be processed. 

https://github.com/microsoft/semantic-kernel/blob/b4bfef1115445ebb08cebfa79ca7a5be7924b3cb/python/semantic_kernel/contents/streaming_chat_message_content.py#L184C9-L196C50

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

We're working towards creating a core set of syntax examples (there are four total). The core examples will be available in all 3 SK languages.

### Description

This PR introduces two (of the four) new kernel syntax examples, which will align with the new kernel examples coming soon for both dotnet and Java. #5784 
- Introduce a custom weather plugin that in conjunction with the core TimePlugin, make use of auto function calling.
- Introduce a kernel syntax example that shows how to integrate with the Microsoft Graph API to create a "restaurant booking." Note: this doesn't actually place a real reservation, but shows how to interact with msgraph.
- Also fixes an issue where the streaming tool call argument formation was broken. Closes #6106

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/README.md">Py3.8 Test Coverage</a><details><summary>Python 3.8 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py">kernel.py</a></td><td>261</td><td>28</td><td>89%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L122">122</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L124">124</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L182-L185">182&ndash;185</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L187-L191">187&ndash;191</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L193-L196">193&ndash;196</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L203">203</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L210-L219">210&ndash;219</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L254-L257">254&ndash;257</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L267-L272">267&ndash;272</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L277">277</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L281-L284">281&ndash;284</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/kernel.py#L466-L467">466&ndash;467</a></td></tr><tr><td colspan="5"><b>semantic_kernel/connectors/ai/open_ai/services</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py">open_ai_chat_completion_base.py</a></td><td>171</td><td>62</td><td>64%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L112">112</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L134-L137">134&ndash;137</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L139-L142">139&ndash;142</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L159-L162">159&ndash;162</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L172-L175">172&ndash;175</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L186-L209">186&ndash;209</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L218-L220">218&ndash;220</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L237-L239">237&ndash;239</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L262">262</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L270">270</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L276-L282">276&ndash;282</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L293-L299">293&ndash;299</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L335-L341">335&ndash;341</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L359">359</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/141ca8b53934ccf94fd9c508c1dd20977b9fda81/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py#L378-L380">378&ndash;380</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5551</b></td><td><b>994</b></td><td><b>82%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.8 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1211 | 11 :zzz: | 0 :x: | 0 :fire: | 24.335s :stopwatch: |

**Comment by markwallace-microsoft** [comment]
<a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/README.md">Py3.10 Test Coverage</a><details><summary>Python 3.10 Test Coverage Report • </summary><table><tr><th>File</th><th>Stmts</th><th>Miss</th><th>Cover</th><th>Missing</th></tr><tbody><tr><td colspan="5"><b>semantic_kernel</b></td></tr><tr><td>&nbsp; &nbsp;<a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py">kernel.py</a></td><td>261</td><td>28</td><td>89%</td><td><a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L122">122</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L124">124</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L182-L185">182&ndash;185</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L187-L191">187&ndash;191</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L193-L196">193&ndash;196</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L203">203</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L210-L219">210&ndash;219</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L254-L257">254&ndash;257</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L267-L272">267&ndash;272</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L277">277</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L281-L284">281&ndash;284</a>, <a href="https://github.com/microsoft/semantic-kernel/blob/bad1cba514be89e3e55f564a25735da46253e276/python/semantic_kernel/kernel.py#L466-L467">466&ndash;467</a></td></tr><tr><td><b>TOTAL</b></td><td><b>5838</b></td><td><b>1038</b></td><td><b>82%</b></td><td>&nbsp;</td></tr></tbody></table></details>

## Python 3.10 Unit Test Overview
| Tests | Skipped | Failures | Errors | Time |
| ----- | ------- | -------- | -------- | ------------------ |
| 1257 | 1 :zzz: | 0 :x: | 0 :fire: | 13.167s :stopwatch: |

File: python/semantic_kernel/kernel.py [hunk 1]
```diff
@@ -346,6 +346,72 @@ async def invoke_prompt(
...
+
+        Returns:
+            AsyncIterable[StreamingContentMixin]: The content of the stream of the last function provided.
+        """
+        if not arguments:
+            arguments = KernelArguments(**kwargs)
+        if not prompt:
+            raise TemplateSyntaxError("The prompt is either null or empty.")
+
+        from semantic_kernel.functions.kernel_function_from_prompt import KernelFunctionFromPrompt
```
Comment (juliomenendez) [inline python/semantic_kernel/kernel.py:382]: Does this import needs to be here? Could it be at the module level?

**Review by juliomenendez — APPROVED** [review]
Looking good! A couple comments but good otherwise.

File: python/semantic_kernel/kernel.py (hunk 1 above)
Comment (moonbox3) [reply python/semantic_kernel/kernel.py:382]: For typing, it does need to be here.

File: python/samples/kernel-syntax-examples/resources/bookings_plugin/bookings_plugin.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                f"{appointment.service_location.display_name} on {appointment.start_date_time.date_time} with id: {appointment.id}"  # noqa: E501
+                for appointment in appointments.value
+            ]
+        )
+
+    @kernel_function(name="cancel_reservation", description="Cancel a reservation")
+    async def cancel_reservation(
+        self,
+        reservation_id: Annotated[str, "The ID of the reservation"],
+        # self, reservation_id: Annotated[str, "The ID of the reservation"]
```
[the remark this answers is not in the document]
Comment (moonbox3) [reply python/samples/kernel-syntax-examples/resources/bookings_plugin/bookings_plugin.py:129]: I can remove, thanks.

File: python/samples/kernel-syntax-examples/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Copyright (c) Microsoft. All rights reserved.
```
[the remark this answers is not in the document]
Comment (moonbox3) [reply python/samples/kernel-syntax-examples/__init__.py]: I was playing around with importing some resources, but I think I can remove.

**Review by juliomenendez — APPROVED** [review]

**Pull request merged**